src/auxiliary/auxiliary.jl

```diff
@@ -316,7 +316,15 @@ function download(src_url, file_path)
     # Note that `mpi_isroot()` is also `true` if running
     # in serial (without MPI).
     if mpi_isroot()
-        isfile(file_path) || Downloads.download(src_url, file_path)
+        if !isfile(file_path)
```

Comment:
Maybe add some additional info to the docstring above (sorry, it was too far away to attach it directly to the correct position):
```julia
#=
[...] avoids race conditions when multiple MPI ranks are used.
Furthermore, when run as part of a GitHub Action, it uses
token-authenticated downloads to avoid GitHub's rate limiting
for unauthenticated HTTP request.
=#
```